sql/set.go

```diff
@@ -41,12 +41,22 @@ func (p *planner) Set(n *parser.Set) (planNode, error) {
 		if err != nil {
 			return nil, err
 		}
-		if s, ok := val.(parser.DString); ok {
-			p.session.Database = string(s)
-		} else if !ok {
+
+		s, ok := val.(parser.DString)
+		if !ok {
 			return nil, fmt.Errorf("database: requires a single string value: %s is a %s",
 				n.Values[0], val.Type())
```

Comment:
Oh yeah, good point. In fact, I think I made that same observation when this code was written. Perhaps I should add a comment so I don't forget again.